File: lib/data/packets/message_packet.dart
import 'package:freezed_annotation/freezed_annotation.dart';

import '../models/message_model.dart';

part 'message_packet.freezed.dart';
part 'message_packet.g.dart';

@freezed
class MessagePacket with _$MessagePacket {
  const MessagePacket._();

  const factory MessagePacket(
      {required String to,
      required MessageModel message,
      required String chatId}) = _MessagePacket;

  factory MessagePacket.fromJson(Map<String, dynamic> json) =>
      _$MessagePacketFromJson(json);
}

File: backend/src/chats/message_packet.ts
import TelleoMessageModel from "../api/chats/data/telleo_message_model";

export default class MessagePacket {
  constructor(
    public readonly message: TelleoMessageModel,
    public readonly to: string,
    public readonly chatId: string
  ) {}

  static fromJson(json: any): MessagePacket {
    return new MessagePacket(
      TelleoMessageModel.fromJson(json.message),
      json.to,
      json.chatId
    );
  }
}
